### DISCUSSION TITLE
隐藏域名后缀，兼容 v1 版本的 url

### DISCUSSION THREAD

**Discussion by ichenhe** [lsky-org/lsky-pro#389] [分享]
v2 本地图片强制添加域名后缀，若从 v1 迁移则自动添加 `i` 后缀，导致已经发布的图片全部失效。

这显然是不可接受的... 😞  其实... 可以利用 web 服务器的 url rewrite 功能来干掉该死的后缀 😄 

以 nginx 为例，只要在 host 配置文件中添加下面的代码：

```nginx
location ~ '^/\d{4}/\d{2}/\d{2}' {
  rewrite ^(.*)$ /i/$1 break;
}
```

完美搞定！（代码里的 `i` 换成你自己设置的后缀，比如 `uploads`）

原理：本地图片的目录格式为 `年/月/日/文件名`，所以我们只要匹配这个格式的请求，在其前面添加上前缀就行了。

**Comment by 0xxb** [comment]
good 👍

**Comment by akatsukiro** [comment]
我的图片好像都没添加/i，挺舒服的。

**Reply by ichenhe** [reply]
你不是本地存储吧

**Reply by akatsukiro** [reply]
有部分是本地存储。链接没变。所以我也感觉挺奇怪的。
